py/torch_tensorrt/dynamo/runtime/_PythonTorchTensorRTModule.py

```diff
@@ -145,15 +144,23 @@ def __getstate__(self) -> Dict[str, Any]:
         state = self.__dict__.copy()
         state["engine"] = bytearray(self.engine.serialize())
         state.pop("context", None)
+        state.pop("input_dtypes", None)
+        state.pop("input_shapes", None)
+        state.pop("output_dtypes", None)
+        state.pop("output_shapes", None)
+        state.pop("active_stream", None)
+        state.pop("target_device_properties", None)
         return state
 
     def __setstate__(self, state: Dict[str, Any]) -> None:
         logger = trt.Logger()
         runtime = trt.Runtime(logger)
-        state["engine"] = runtime.deserialize_cuda_engine(state["engine"])
         self.__dict__.update(state)
+        self.target_device_properties = torch.cuda.get_device_properties(
```

Comment:
The deserialization should make sure that the gpu matches the gpu the engine was built on instead of just changing it